### ISSUE TITLE
Icons misplaced

### ISSUE THREAD

**Issue by felixschndr** [openhab/openhab-android#3786]
Disclaimer: There are some open issues about icons but I am not sure weather any of these covers the following issue. If so please close this one.

### Actual behaviour
Since a recent version of the android app the icons seem to be misplaced sometimes:
![Screenshot_20240721_122802_openHAB](https://github.com/user-attachments/assets/6073ccce-20c8-4c60-a9ce-4459f939b096)
![Screenshot_20240721_122807_openHAB](https://github.com/user-attachments/assets/c7d6955f-7929-44ce-b3b5-4f6d8d958a88)
After scrolling in the window or reloading the icon seem to be placed where they belong.

Clearing the cache does not help.

In the following image I noticed that an icon got a background it should not have:
![Screenshot_20240721_122759_openHAB](https://github.com/user-attachments/assets/3722ef6d-3d37-4463-85a0-0e9f229516cb)
The original image has no white background.

On iOS the icons work as expected:

![IMG_0019](https://github.com/user-attachments/assets/278434a8-7469-43f7-9f8d-19b96b80ceae)

It does not matter whether the connection is local or via the cloud.

### Expected behaviour
The icons should all be aligned the same way.

 
### Steps to reproduce
See code below

#### Client
* Android version: 14
* Device model: Samsung Galaxy S22 Ultra
* App version 3.15.5 (I _think_ it happens since 3.15.0)
* Build flavor :  Full
* Device language: German

#### Server
* Server version: 4.2.0
* Reverse Proxy: Nginx
* Authentication method : None

#### App log
<details>
  <summary>Click to expand</summary>

[---------------.txt](https://github.com/user-attachments/files/16411933/---------------.txt)

</details>

Code for first image:
```txt
        Frame label="Saugen" {
            Switch item=Staubsauger_System_Staubsauger_Steuerung
                mappings=["vacuum"="Start", "pause"="Pause", "dock"="Dock"]
                visibility=[Staubsauger_System_Staubsauger_Status != "Charging"]
            Text item=Staubsauger_System_Staubsauger_Status
                visibility=[!= "Charging"]
            Selection item=Staubsauger_System_Staubsauger_Saugleistung
                mappings=[101="Leise", 102="Normal", 103="Turbo", 104="Maximal"]
            Text item=Staubsauger_System_Raum_Gruppe
                visibility=[Staubsauger_System_Staubsauger_Status == "Charging"]
            {
                Switch item=Staubsauger_System_Raum_16
                Switch item=Staubsauger_System_Raum_23
                Switch item=Staubsauger_System_Raum_17
                Switch item=Staubsauger_System_Raum_20
                Switch item=Staubsauger_System_Raum_18
                Switch item=Staubsauger_System_Raum_19
            }
            Slider item=Staubsauger_System_Reinigungszyklen minValue=1 maxValue=3
                visibility=[Staubsauger_System_Staubsauger_Status == "Charging"]
            Switch item=Staubsauger_System_Start
                label=""
                mappings=[ON="Start"]
                visibility=[Staubsauger_System_Raum_Gruppe != 0]

            Image item=Staubsauger_System_Staubsauger_Karte
                visibility=[!= UNDEF]
....
```

Code for second image:
```txt
        Frame label="Bindings" {
            Switch item=System_System_Binding_AmazonEchoControl
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_AndroidTV
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_ESPHome
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_Nanoleaf
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_Miio
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_Network
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_OpenWeatherMap
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_Hue
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_Shelly
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_Spotify
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
            Switch item=System_System_Binding_Systeminfo
                mappings=["Stop"="STOP", "Restart"="NEUSTART", "Start"="START"]
        }
```

Code for third image:
```txt
        Frame label="Klima" {
            Text item=Klima_System_Durchschnittstemperatur {
                Text item=Klima_Badezimmer_Temperatur
                    label="Badezimmer"
                Text item=Klima_Flur_Temperatur
                    label="Flur"
                Text item=Klima_Kueche_Temperatur
                    label="Küche"
                Text item=Klima_Schlafzimmer_Temperatur
                    label="Schlafzimmer"
            }
            Text item=Klima_System_Aussentemperatur_Zusammengesetzt
        }
```

**Comment by felixschndr** [comment]
Might be related/duplicate to https://github.com/openhab/openhab-android/issues/3773 and https://github.com/openhab/openhab-android/issues/3781

**Comment by maniac103** [comment]
The switch-with-mappings icon issue likely is #3773. The 'white background' one likely is one I also have seen once, but didn't have a chance of investigation yet: It looks like the icon and the loading placeholder ('glowing' frame) are both shown simultaneously.

**Comment by felixschndr** [comment]
Then I guess we can close this as duplicate and/or should create an issue for 
>  It looks like the icon and the loading placeholder ('glowing' frame) are both shown simultaneously.

**Comment by maniac103** [comment]
We can keep this one for that issue.

As far as I can tell, this placeholder (skeleton) + image issue happens due to race conditions when views are reused, because
- `WidgetImageView` puts itself into a `SkeletonLayout`
- which sets the `WidgetImageView` to invisible while the skeleton is shown
- `LabeledItemBaseViewHolder.bind()` also manipulates the visibility flag of the `WidgetImageView` because it's unaware the view is now contained in another view (and the adapter now actually wanted to hide/show that one)

The solution for this, as far as I can tell, is avoiding the magic implicit skeleton creation. Will think about it further, though.

**Comment by maniac103** [comment]
Evidence of speculation above: I've added some debug statements to `WidgetImageView`:

```
[...]
07-29 15:37:36.050 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: setImageUrl(chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540, force true), target size 996, last http://<ohserver>:8080/chart?groups=GroupPoolTemperature&dpi=420&period=3D&service=influxdb&theme=white_transparent&yAxisDecimalPattern=&w=1080&h=540
07-29 15:37:36.053 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: URL http://<ohserver>:8080/chart?groups=GroupPoolTemperature&dpi=420&period=3D&service=influxdb&theme=white_transparent&yAxisDecimalPattern=&w=1080&h=540 -> http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540
07-29 15:37:36.054 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: show skeleton for com.faltenreich.skeletonlayout.SkeletonLayout{48a001f V.ED..... .......D 0,0-996,498 aid=1073742071}, parent is com.faltenreich.skeletonlayout.SkeletonLayout{48a001f V.ED..... .......D 0,0-996,498 aid=1073742071} ... android.widget.FrameLayout{1c5af6c V.E...... .......D 42,32-1038,530 aid=1073742066}
07-29 15:37:36.056 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: set visibility invisible
07-29 15:37:36.056 31925 31925 D WidgetImageView: java.lang.Throwable
07-29 15:37:36.056 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.setVisibility(WidgetImageView.kt:142)
07-29 15:37:36.056 31925 31925 D WidgetImageView:       at com.faltenreich.skeletonlayout.SkeletonLayout.showSkeleton(SkeletonLayout.kt:61)
07-29 15:37:36.056 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.applySkeleton(WidgetImageView.kt:352)
07-29 15:37:36.056 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.doLoad(WidgetImageView.kt:286)
07-29 15:37:36.056 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.setImageUrl(WidgetImageView.kt:114)
07-29 15:37:36.056 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.setImageUrl$default(WidgetImageView.kt:87)
07-29 15:37:36.056 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$ChartViewHolder.bindAfterDataSaverCheck$mobile_fossBetaDebug(WidgetAdapter.kt:1440)
07-29 15:37:36.056 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$HeavyDataViewHolder.bind(WidgetAdapter.kt:522)
[...]
07-29 15:37:36.059 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a IFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: execute request for http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540
07-29 15:37:36.059 31925 31925 I WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a IFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: Refreshing image at http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540, avoidCache true
07-29 15:37:36.062 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a IFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: refresh on attach, request HttpImageRequest(url=http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540, job=null)
07-29 15:37:36.062 31925 31925 I WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a IFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: Refreshing image at http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540, avoidCache false
07-29 15:37:36.090 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{e336de3 V.ED..... ......I. 0,0-0,0 #7f0a030d app:id/widgeticon}: set visibility visible
07-29 15:37:36.090 31925 31925 D WidgetImageView: java.lang.Throwable
07-29 15:37:36.090 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.setVisibility(WidgetImageView.kt:142)
07-29 15:37:36.090 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$LabeledItemBaseViewHolder.bind(WidgetAdapter.kt:2005)
07-29 15:37:36.090 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$HeavyDataViewHolder.bind(WidgetAdapter.kt:516)
07-29 15:37:36.090 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:265)
07-29 15:37:36.090 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:127)
07-29 15:37:36.090 31925 31925 D WidgetImageView:       at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7747)
07-29 15:37:36.090 31925 31925 D WidgetImageView:       at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7847)
[...]
07-29 15:37:36.156 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a IFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: starting download of http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540&random=-677746331
[...]
07-29 15:37:36.240 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a IFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: set visibility visible
07-29 15:37:36.240 31925 31925 D WidgetImageView: java.lang.Throwable
07-29 15:37:36.240 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.setVisibility(WidgetImageView.kt:142)
07-29 15:37:36.240 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$HeavyDataViewHolder.showDataSaverPlaceholderIfNeeded(WidgetAdapter.kt:2009)
07-29 15:37:36.240 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$HeavyDataViewHolder.bind(WidgetAdapter.kt:521)
07-29 15:37:36.240 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:265)
07-29 15:37:36.240 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:127)
07-29 15:37:36.240 31925 31925 D WidgetImageView:       at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7747)
07-29 15:37:36.240 31925 31925 D WidgetImageView:       at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7847)
[...]
07-29 15:37:36.242 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: setImageUrl(chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540, force true), target size 996, last http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540
07-29 15:37:36.242 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ......ID 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: URL equals last URL, active true
[...]
07-29 15:37:36.307 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ........ 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: set visibility visible
07-29 15:37:36.307 31925 31925 D WidgetImageView: java.lang.Throwable
07-29 15:37:36.307 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.setVisibility(WidgetImageView.kt:142)
07-29 15:37:36.307 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$HeavyDataViewHolder.showDataSaverPlaceholderIfNeeded(WidgetAdapter.kt:2009)
07-29 15:37:36.307 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$HeavyDataViewHolder.bind(WidgetAdapter.kt:521)
07-29 15:37:36.307 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:265)
07-29 15:37:36.307 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:127)
07-29 15:37:36.307 31925 31925 D WidgetImageView:       at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7747)
[...]
07-29 15:37:36.309 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ........ 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: setImageUrl(chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540, force true), target size 996, last http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540
07-29 15:37:36.309 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ........ 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: URL equals last URL, active true
[...]
07-29 15:37:36.393 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ........ 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: set visibility visible
07-29 15:37:36.393 31925 31925 D WidgetImageView: java.lang.Throwable
07-29 15:37:36.393 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.setVisibility(WidgetImageView.kt:142)
07-29 15:37:36.393 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$HeavyDataViewHolder.showDataSaverPlaceholderIfNeeded(WidgetAdapter.kt:2009)
07-29 15:37:36.393 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter$HeavyDataViewHolder.bind(WidgetAdapter.kt:521)
07-29 15:37:36.393 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:265)
07-29 15:37:36.393 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.WidgetAdapter.onBindViewHolder(WidgetAdapter.kt:127)
07-29 15:37:36.393 31925 31925 D WidgetImageView:       at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7747)
07-29 15:37:36.393 31925 31925 D WidgetImageView:       at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7847)
07-29 15:37:36.393 31925 31925 D WidgetImageView:       at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6646)
[...]
07-29 15:37:38.095 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ........ 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: loaded bitmap android.graphics.Bitmap@f7b9364 for http://<ohserver>:8080/chart?groups=GroupSolarProduction&dpi=420&period=1D&theme=white_transparent&legend=true&yAxisDecimalPattern=&w=1080&h=540
07-29 15:37:38.096 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ........ 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: hide skeleton for com.faltenreich.skeletonlayout.SkeletonLayout{48a001f V.ED..... ......ID 0,0-996,498 aid=1073742071}
07-29 15:37:38.098 31925 31925 D WidgetImageView: org.openhab.habdroid.ui.widget.WidgetImageView{133de1a VFED..C.. ........ 0,0-996,498 #7f0a030c app:id/widget_content aid=1073742065}: set visibility visible
07-29 15:37:38.098 31925 31925 D WidgetImageView: java.lang.Throwable
07-29 15:37:38.098 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.setVisibility(WidgetImageView.kt:142)
07-29 15:37:38.098 31925 31925 D WidgetImageView:       at com.faltenreich.skeletonlayout.SkeletonLayout.showOriginal(SkeletonLayout.kt:50)
07-29 15:37:38.098 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.removeSkeleton(WidgetImageView.kt:357)
07-29 15:37:38.098 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.applyLoadedBitmap(WidgetImageView.kt:322)
07-29 15:37:38.098 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView.access$applyLoadedBitmap(WidgetImageView.kt:50)
07-29 15:37:38.098 31925 31925 D WidgetImageView:       at org.openhab.habdroid.ui.widget.WidgetImageView$HttpImageRequest$execute$1.invokeSuspend(WidgetImageView.kt:406)
```

One can see clearly that
- a download (here: chart) is started
- the view is rebound a few times while the download is running
- on rebind, the image is made visible (while it shouldn't be at that time)
- on download finish, the correct visibility toggle happens

**Comment by maniac103** [comment]
Fixed with #3787
